src/components/structures/MessagePanel.tsx

```diff
@@ -239,7 +243,8 @@ export default class MessagePanel extends React.Component<IProps, IState> {
         };
 
         // Cache hidden events setting on mount since Settings is expensive to
-        // query, and we check this in a hot code path.
+        // query, and we check this in a hot code path. This is also cached in
+        // our RoomContext, however we still need a fallback for roomless MessagePanels.
         this.showHiddenEventsInTimeline = SettingsStore.getValue("showHiddenEventsInTimeline");
```

Comment:
```suggestion
        this.showHiddenEventsInTimeline = context?.showHiddenEventsInTimeline
            ?? SettingsStore.getValue("showHiddenEventsInTimeline");
```

Reply:
then replace later instances of `this.context?.showHiddenEventsInTimeline` with `this.showHiddenEventsInTimeline`